src/renderer/components/+workloads-pods/pod-container-env.tsx

```diff
@@ -89,19 +93,52 @@ export const ContainerEnvironment = observer((props: Props) => {
 
   const renderEnvFrom = () => {
     const envVars = envFrom.map(vars => {
-      if (!vars.configMapRef || !vars.configMapRef.name) return;
-      const configMap = configMapsStore.getByName(vars.configMapRef.name, namespace);
+      if (vars.configMapRef?.name) {
+        return renderEnvFromConfigMap(vars.configMapRef.name);
+      } else if (vars.secretRef?.name ) {
+        return renderEnvFromSecret(vars.secretRef.name);
+      }
+    });
+
+    return _.flatten(envVars);
+  };
 
-      if (!configMap) return;
+  const renderEnvFromConfigMap = (configMapName: string) => {
+    const configMap = configMapsStore.getByName(configMapName, namespace);
 
-      return Object.entries(configMap.data).map(([name, value]) => (
-        <div className="variable" key={name}>
-          <span className="var-name">{name}</span>: {value}
+    if (!configMap) return;
+
+    return Object.entries(configMap.data).map(([name, value]) => (
+      <div className="variable" key={name}>
+        <span className="var-name">{name}</span>: {value}
+      </div>
+    ));
+  };
+
+  const renderEnvFromSecret = (secretName: string) => {
+    const secret = secretsStore.getByName(secretName, namespace);
+
+    if (!secret) return;
```

Comment:
Shouldn't we render the name only here?

Reply:
Maybe having `envFrom` section in pod details where secretKeyRef is rendered, so not to mix it with environment variables. But that can be done in follow-up PR if we feel it's a good idea.